### PR TITLE
[READY] borg changes and new module that makes borgs contribute on the research

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl: 
add: new research module upgrade for borgs: it increases idle power consuption and generates techpoints for around 400 / min (placeholder)
tweak: changed prices for cyborg upgrades to be more reasonable
tweak: the mediborg extra chem upgrade is now in 1 single module instead of 2
/:cl:

### CODE REVIEW

**Comment by daklaj** [comment]
disgusting

**Comment by Tlaltecuhtli** [comment]
yog players begone

**Comment by MMMiracles** [comment]
nanotrasen is hurting for cash so they're installing cryptominers on all cyborgs to generate extra revenue

**Comment by GuyonBroadway** [comment]
WARNING! FUN QUOTA HAS BEEN REACHED! YOU MUST WAIT _"40"_ MINUTES BEFORE YOU MAY RESUME HAVING FUN!

**Comment by davethwave** [comment]
Seems good to me.

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 1]
```diff
@@ -665,34 +665,25 @@
...
 /datum/design/borg_upgrade_expandedsynthesiser
 	name = "Cyborg Upgrade (Hypospray Expanded Synthesiser)"
 	id = "borg_upgrade_expandedsynthesiser"
 	build_type = MECHFAB
 	build_path = /obj/item/borg/upgrade/hypospray/expanded
-	materials = list(MAT_METAL=15000, MAT_GLASS=15000, MAT_PLASMA=5000)
-	construction_time = 120
-	category = list("Cyborg Upgrade Modules")
-
-/datum/design/borg_upgrade_highstrengthsynthesiser
```
Comment (vuonojenmustaturska) [inline code/modules/research/designs/mechfabricator_designs.dm]: why did you remove this

Comment (davethwave) [reply code/modules/research/designs/mechfabricator_designs.dm]: He said he was merging it with the other synthesiser upgrade in the description.

Comment (Jalleo) [reply code/modules/research/designs/mechfabricator_designs.dm]: There are a few other changes that has no mention. This one does but the ones that are getting changed are mainly cheaper.

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 2]
```diff
@@ -665,34 +665,25 @@
...
-	name = "Cyborg Upgrade (Hypospray High-Strength Synthesiser)"
-	id = "borg_upgrade_highstrengthsynthesiser"
-	build_type = MECHFAB
-	build_path = /obj/item/borg/upgrade/hypospray/high_strength
-	materials = list(MAT_METAL=15000, MAT_GLASS=15000, MAT_PLASMA=10000, MAT_URANIUM=5000)
-	construction_time = 120
+	materials = list(MAT_METAL = 10000, MAT_GLASS = 5000, MAT_PLASMA = 5000, MAT_URANIUM = 5000)
```
Comment (Jalleo) [inline code/modules/research/designs/mechfabricator_designs.dm]: This needs increasing NOT decreasing.

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 3]
```diff
@@ -665,34 +665,25 @@
...
+	construction_time = 80
 	category = list("Cyborg Upgrade Modules")
 
 /datum/design/borg_upgrade_piercinghypospray
 	name = "Cyborg Upgrade (Piercing Hypospray)"
 	id = "borg_upgrade_piercinghypospray"
 	build_type = MECHFAB
 	build_path = /obj/item/borg/upgrade/piercing_hypospray
-	materials = list(MAT_METAL=15000, MAT_GLASS=15000, MAT_TITANIUM=10000, MAT_DIAMOND=5000)
+	materials = list(MAT_METAL = 5000, MAT_GLASS = 5000, MAT_TITANIUM = 5000, MAT_DIAMOND = 4000)
```
Comment (Jalleo) [inline code/modules/research/designs/mechfabricator_designs.dm]: Flat out a power gaming motive on this by making it cheaper.

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 4]
```diff
@@ -701,46 +692,55 @@
 	id = "borg_upgrade_defibrillator"
 	build_type = MECHFAB
 	build_path = /obj/item/borg/upgrade/defib
-	materials = list(MAT_METAL=15000, MAT_GLASS=15000, MAT_SILVER=10000, MAT_GOLD=10000, MAT_TITANIUM=5000, MAT_DIAMOND=5000)
-	construction_time = 120
+	materials = list(MAT_METAL = 8000, MAT_GLASS = 5000, MAT_SILVER = 4000, MAT_GOLD = 3000)
```
Comment (Jalleo) [inline code/modules/research/designs/mechfabricator_designs.dm]: Suitable since its vs having something else but primary question for my approval as a anti power gamer contributor does this need two or one slot on a borg?

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 5]
```diff
@@ -701,46 +692,55 @@
...
-	
+
 /datum/design/borg_upgrade_surgicalprocessor
 	name = "Cyborg Upgrade (Surgical Processor)"
 	id = "borg_upgrade_surgicalprocessor"
 	build_type = MECHFAB
 	build_path = /obj/item/borg/upgrade/processor
-	materials = list(MAT_METAL=15000, MAT_GLASS=15000, MAT_SILVER=10000)
-	construction_time = 120
+	materials = list(MAT_METAL = 5000, MAT_GLASS = 400, MAT_SILVER = 4000)
```
Comment (Jalleo) [inline code/modules/research/designs/mechfabricator_designs.dm]: Maybe reduced too much make it half instead of 60% cheaper.

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 6]
```diff
@@ -701,46 +692,55 @@
...
 	category = list("Cyborg Upgrade Modules")
 
 /datum/design/borg_upgrade_trashofholding
 	name = "Cyborg Upgrade (Trash Bag of Holding)"
 	id = "borg_upgrade_trashofholding"
 	build_type = MECHFAB
 	build_path = /obj/item/borg/upgrade/tboh
-	materials = list(MAT_METAL=10000, MAT_GOLD=1500, MAT_URANIUM=250, MAT_PLASMA=1500)
-	construction_time = 120
+	materials = list(MAT_GOLD = 2000, MAT_URANIUM = 1000)
```
Comment (Jalleo) [inline code/modules/research/designs/mechfabricator_designs.dm]: Maybe too cheap it should require slightly more.  Have it cost a sheet of metal still.

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 7]
```diff
@@ -575,25 +575,25 @@
...
+	construction_time = 80
 	category = list("Cyborg Upgrade Modules")
 
 /datum/design/borg_upgrade_vtec
 	name = "Cyborg Upgrade (VTEC Module)"
 	id = "borg_upgrade_vtec"
 	build_type = MECHFAB
 	build_path = /obj/item/borg/upgrade/vtec
-	materials = list(MAT_METAL=80000 , MAT_GLASS=6000 , MAT_URANIUM= 5000)
+	materials = list(MAT_METAL = 25000 , MAT_GLASS = 6000 , MAT_URANIUM = 5000)
```
Comment (Jalleo) [inline code/modules/research/designs/mechfabricator_designs.dm]: Requires more metal. This is the speed upgrade its powerful.

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 8]
```diff
@@ -575,25 +575,25 @@
...
 	category = list("Cyborg Upgrade Modules")
 
 /datum/design/borg_upgrade_restart
 	name = "Cyborg Upgrade (Emergency Reboot Board)"
 	id = "borg_upgrade_restart"
 	build_type = MECHFAB
 	build_path = /obj/item/borg/upgrade/restart
-	materials = list(MAT_METAL=60000 , MAT_GLASS=5000)
-	construction_time = 120
+	materials = list(MAT_METAL = 20000 , MAT_GLASS = 5000)
```
Comment (Jalleo) [inline code/modules/research/designs/mechfabricator_designs.dm]: Might become used more.

**Review by Jalleo — CHANGES_REQUESTED** [review]
Too many unstated changes that are useful to cyborgs.

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 2 above)
Comment (octareenroon91) [reply code/modules/research/designs/mechfabricator_designs.dm]: Uranium is probably still the limiting material here, and it's unchanged.

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 3 above)
Comment (octareenroon91) [reply code/modules/research/designs/mechfabricator_designs.dm]: Agreeing here. Diamond is the rarest material in this list, and it especially shouldn't go from 2+ sheets to exactly 2 sheets.

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 9]
```diff
@@ -636,3 +630,32 @@
...
+		if(U)
+			to_chat(user, "<span class='warning'>This unit is already equipped with a research module.</span>")
+			return FALSE
+		else
+			START_PROCESSING(SSobj, src)
+
+/obj/item/borg/upgrade/research/process()
+	if(cyborg.cell.charge > 100)
+		cyborg.cell.use(100)
+		SSresearch.science_tech.add_point_list(list(TECHWEB_POINT_TYPE_GENERIC = 50))
```
Comment (XDTM) [inline code/game/objects/items/robot/robot_upgrades.dm]: With the SSobj tick rate this amounts to 1500 per minute per borg

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 10]
```diff
@@ -636,3 +630,32 @@
...
+	var/mob/living/silicon/robot/cyborg
+
+/obj/item/borg/upgrade/research/action(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(.)
+		var/obj/item/borg/upgrade/research/U = locate() in R
+		cyborg = R
+		if(U)
+			to_chat(user, "<span class='warning'>This unit is already equipped with a research module.</span>")
+			return FALSE
```
Comment (XDTM) [inline code/game/objects/items/robot/robot_upgrades.dm]: Make it prevent insertion if the borg already has a research module

File: code/game/objects/items/robot/robot_upgrades.dm (hunk 9 above)
Comment (Tlaltecuhtli) [reply code/game/objects/items/robot/robot_upgrades.dm]: still have to do these numbers

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 8 above)
Comment (XDTM) [reply code/modules/research/designs/mechfabricator_designs.dm]: Reasonable considering cloning is completely free

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 11]
```diff
@@ -575,25 +575,25 @@
...
+	construction_time = 80
 	category = list("Cyborg Upgrade Modules")
 
 /datum/design/borg_upgrade_vtec
 	name = "Cyborg Upgrade (VTEC Module)"
 	id = "borg_upgrade_vtec"
 	build_type = MECHFAB
 	build_path = /obj/item/borg/upgrade/vtec
-	materials = list(MAT_METAL=80000 , MAT_GLASS=6000 , MAT_URANIUM= 5000)
+	materials = list(MAT_METAL = 35000 , MAT_GLASS = 12000 , MAT_URANIUM = 10000)
```
Comment (Jalleo) [inline code/modules/research/designs/mechfabricator_designs.dm]: This seems more suitable now but glass is cheaper than metal. The only question is will miners gather it up more.

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 6 above)
Comment (Tlaltecuhtli) [reply code/modules/research/designs/mechfabricator_designs.dm]: similar price to handheld trashbags

File: code/game/objects/items/robot/robot_upgrades.dm (hunk 10 above)
Comment (Tlaltecuhtli) [reply code/game/objects/items/robot/robot_upgrades.dm]: what

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 4 above)
Comment (Tlaltecuhtli) [reply code/modules/research/designs/mechfabricator_designs.dm]: similar price to defib

File: code/game/objects/items/robot/robot_upgrades.dm (hunk 10 above)
Comment (XDTM) [reply code/game/objects/items/robot/robot_upgrades.dm]: The way you set it up, if i read it correctly, still consumes the module even if the borg already has one.

Comment (Tlaltecuhtli) [reply code/game/objects/items/robot/robot_upgrades.dm]: it drops it on the floor gonna test it again to be 100% sure

Comment (Tlaltecuhtli) [reply code/game/objects/items/robot/robot_upgrades.dm]: tested it

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 4 above)
Comment (XDTM) [reply code/modules/research/designs/mechfabricator_designs.dm]: This is a better version since borgs don't have backpacks to sacrifice to hold it

**Comment by ExcessiveUseOfCobblestone** [comment]
I thought the nanite PR said no passive gen?

That said, is byond the punishment role meme and more of a "You are to literally sit in the recharger for this to function". Add that to the fact asimov is the sub role and now you have people unable to play the game because lol gotta get the points!

At high cap this goes off in 100 iterations. Normal batteries last for 50.

**Comment by Tlaltecuhtli** [comment]
ask your masters to upgrade your cell/ borg chargers

**Comment by ExcessiveUseOfCobblestone** [comment]
Why would they?

You can sit in the recharger indefinitely and, per asimov law 2, you'd have to.

If you refuse to (rogue), then you only last a minute or two assuming all you do is run around.

This can be used maliciously just as much as it can be beneficial.

**Comment by XDTM** [comment]
just get a couple borgs to go braindead inside rechargers and you have your point generators going; make more posibrains and borgs and you can have a borg server farm going

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 12]
```diff
@@ -636,3 +630,32 @@
...
+		else
+			START_PROCESSING(SSobj, src)
+
+/obj/item/borg/upgrade/research/process()
+	if(cyborg.cell.charge > 150)
+		cyborg.cell.use(150)
+		SSresearch.science_tech.add_point_list(list(TECHWEB_POINT_TYPE_GENERIC = 15))
```
Comment (ma44) [inline code/game/objects/items/robot/robot_upgrades.dm]: Should probably check for a ckey/client just in case, otherwise you could get cyborg's posibrains to ghost so you can make a ton of these

**Comment by ExcessiveUseOfCobblestone** [comment]
@optimumtact thoughts?

**Comment by ma44** [comment]
You could also just probably make the RnD servers take up power for rnd points so you don't have to make someone's robotic life miserable hanging around in a recharger for a long time

**Comment by Tlaltecuhtli** [comment]
its supposed to increase the idle power usage not making the borg have to sit in the recharger all round

File: code/modules/research/techweb/all_nodes.dm [hunk 13]
```diff
@@ -273,7 +273,7 @@
 	display_name = "Cyborg Upgrades: Medical"
 	description = "Medical upgrades for cyborgs."
 	prereq_ids = list("adv_biotech")
-	design_ids = list("borg_upgrade_defibrillator", "borg_upgrade_piercinghypospray", "borg_upgrade_highstrengthsynthesiser", "borg_upgrade_expandedsynthesiser", "borg_upgrade_pinpointer", "borg_upgrade_surgicalprocessor")
+	design_ids = list("borg_upgrade_defibrillator", "borg_upgrade_piercinghypospray", ""borg_upgrade_expandedsynthesiser", "borg_upgrade_pinpointer", "borg_upgrade_surgicalprocessor")
```
Comment (Carbonhell) [inline code/modules/research/techweb/all_nodes.dm]: There's an excess " that fucks everything up ""borg_upgrade_expandedsynthesiser"

**Comment by GunHog** [comment]
The borg player can turn it off, right?

**Comment by Tlaltecuhtli** [comment]
splitting up pr
